controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -315,6 +315,13 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 		machine.Spec.NodeDeletionTimeout = kcp.Spec.MachineTemplate.NodeDeletionTimeout
 	}
 
+	// In case this machine is being created as a consequence of a remediation, then add an annotation
+	// tracking the name of the machine we are remediating for.
+	// NOTE: This is required in order to track remediation retries.
+	if v, ok := kcp.Annotations[controlplanev1.RemediationInProgressAnnotation]; ok && v == "true" {
```

Comment:
I couldn't find any (prod) code which sets the annotation to "true".

Reply:
Which means we don't set the `RemediationForAnnotation` at the moment, which means `if remediationFor, ok := machineToBeRemediated.Annotations[controlplanev1.RemediationForAnnotation]; ok {` in remediation.go shouldn't work at the moment.

Reply:
Yes, but it is not set to "true". so `v == "true"` will be always false

Reply:
this logic changed